R/match_pc_gpprac.R

```diff
@@ -0,0 +1,268 @@
+    dplyr::inner_join(data_gpprac,
+      ggprac_lookup %>%
+        dplyr::select(-(
+          2:dplyr::last_col()
+        )),
+      by = "gpprac"
+    ) %>% dplyr::mutate(gpprac_match = 1),
+    # For the rows that do not match, give value of gpprac_match = 0
+    dplyr::anti_join(data_gpprac,
+      ggprac_lookup %>%
+        dplyr::select(-(
+          2:dplyr::last_col()
+        )),
+      by = "gpprac"
+    ) %>% dplyr::mutate(gpprac_match = 0)
+  )
+
+  data_gpprac_match_info <- data_gpprac_match %>%
+    dplyr::mutate(gpprac_match = dplyr::if_else(
+      gpprac %in% c(99942, 99957, 99961, 99976, 99981, 99995, 99999),
+      0,
+      gpprac_match
+    )) %>%
+    dtplyr::lazy_dt() %>%
+    dplyr::group_by(chi) %>%
+    dplyr::summarise(all_match = mean(gpprac_match)) %>%
+    dplyr::ungroup() %>%
+    tibble::as_tibble()
+
+  data_gpprac_2 <-
+    dplyr::left_join(data_gpprac_match, data_gpprac_match_info, by = "chi") %>%
+    dplyr::mutate(potentially_fixable = !is_missing(chi) &
+      (all_match != 0 & all_match != 1)) %>%
+    dplyr::filter(potentially_fixable == TRUE) %>%
+    dplyr::group_by(chi) %>%
+    # Arrange by one of the keydates so the most recent gpprac is at the top even if it's NA
+    # Then fill the values of gpprac upwards so the NA is filled in
+    dplyr::arrange(desc(is.na(gpprac)), desc(keydate2_dateformat), .by_group = TRUE) %>%
+    tidylog::mutate(gpprac = dplyr::if_else(postcode == 9999, NA_real_, gpprac)) %>%
+    tidyr::fill(postcode, .direction = "up") %>%
+    dplyr::ungroup()
+
+  data_gpprac_3 <- dplyr::bind_rows(
+    data_gpprac_2,
+    dplyr::left_join(data_gpprac_match, data_gpprac_match_info, by = "chi") %>%
+      dplyr::mutate(
+        potentially_fixable = !is_missing(chi) &
+          (all_match != 0 & all_match != 1)
+      ) %>%
+      dplyr::filter(potentially_fixable == FALSE)
+  ) %>%
+    dplyr::select(-all_match, -potentially_fixable, -gpprac_match)
+
+  data_gpprac_4 <- dplyr::bind_rows(
+    # First, get all the rows that do match, and give the variable gpprac_match = 1
+    dplyr::inner_join(
+      data_gpprac_3,
+      ggprac_lookup %>%
+        dplyr::select(gpprac, hbpraccode, cluster),
+      by = "gpprac"
+    ) %>% dplyr::mutate(gpprac_match = 1),
+    # For the rows that do not match, give value of gpprac_match = 0
+    dplyr::anti_join(
+      data_gpprac_3,
+      ggprac_lookup %>%
+        dplyr::select(gpprac, hbpraccode, cluster),
+      by = "gpprac"
+    ) %>% dplyr::mutate(gpprac_match = 0)
+  ) %>%
+    dplyr::mutate(
+      hbpraccode = dplyr::if_else(gpprac_match == 0, hbpraccode_old, hbpraccode),
+      gpprac = dplyr::if_else(gpprac_match == 0 &
+        is_missing(hbpraccode), NA_real_, gpprac)
+    )
+
+  # Here is a section in SPSS for adding value labels. I think those may be redundant?
+  # * All geographies should now have a label and be from the 2018 (or dummy) geographies.
+
+  # save the data to the folder
+  # write_rds()
+
+  # return()
```

Comment:
Is this function supposed to return a data.frame or simply write out the data?

Reply:
It probably doesn't need to write anything out. Just return the data. We had to write out data really often in SPSS but I don't think it's required in R.